---
title: "Microbiome Section Report"
author: "Angelica Torres"
date: "2024-05-02"
output: pdf_document
---

# Loading required packages
```{r}
library(dada2)
```

# Load Biocmanager
```{r}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("dada2", version = "3.18")
```

# Load sequences
```{r}
path <- "sequences"
list.files(path)
```

# Forward and reverse reads
```{r}


```

# Read in file names
```{r}
# fastq filenames have format: SAMPLENAME_R1_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))

# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```

# Inspect read quality
```{r}
plotQualityProfile(fnFs[1:2])
```

# Filter and Trim
```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))

names(filtFs) <- sample.names

# quality decreases at 200
out <- filterAndTrim(fnFs, filtFs, truncLen=c(200),
              maxN=0, maxEE=c(2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE
head(out)
```

# Learn error rates
```{r}
errF <- learnErrors(filtFs, multithread=FALSE)
plotErrors(errF, nominalQ=TRUE)
```

# Sample Inference
```{r}
dadaFs <- dada(filtFs, err=errF, multithread=FALSE)
```

# Create sequence table
```{r}
seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)

# Inspect distribution of sequence lengths
table(nchar(getSequences(seqtab)))
```

# remove chimeras
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=FALSE, verbose=TRUE)
dim(seqtab.nochim)
```

# track reads through pipeline
```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(out, sapply(dadaFs, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "nonchim")
rownames(track) <- sample.names
head(track)
```

# Save seqtab with nochim in file
```{r}
save(seqtab.nochim, file="RData/seqtab.nochim.RData")
```

# Load seqtab.nochim to start here
```{r}
load("RData/seqtab.nochim.Rdata")
```

# Assign Taxonomy
```{r}
# download the Silva species database from https://zenodo.org/records/4587955 
taxa <- assignTaxonomy(seqtab.nochim, "silva_nr99_v138.1_wSpecies_train_set.fa.gz")
```

# Save taxonomy as a file
```{r}
save(taxa, file = "RData/taxa.RData")
```
